### DISCUSSION TITLE
OSC near-RT RIC works with srsRAN

### DISCUSSION THREAD

**Discussion by Summer-GY** [srsran/srsRAN_Project#520] [General Help] [answered]
I've noticed that FlexRIC and srsRAN have many issues that prevent them from operating normally. I would like to ask if the OSC near-real time RIC can be used normally with the RAN Control xApp to perform some actions, such as Radio Resource Allocation, to control the RAN?

**Comment by P4sca1** [comment]
I am working with the OSC I-Release and srsRAN. The connection process (E2SetupRequest and E2SetupResponse) is working. I tried to deploy the kpimon-go xApp for basic metric reporting, but the app crashes when trying to send the subscription request.

**Reply by mohmd-abzd** [reply]
Hi there.

Where can I find the subscription request message? And how are you deploying your xApp?

**Reply by Ali78k** [reply]
> I am working with the OSC I-Release and srsRAN. The connection process (E2SetupRequest and E2SetupResponse) is working. I tried to deploy the kpimon-go xApp for basic metric reporting, but the app crashes when trying to send the subscription request.

dear @P4sca1 , I've experienced the exact same issue. Were you able to resolve it?

**Comment by Summer-GY** [comment]
On the tutorial, it says Control Service Style 2 is supported. So, which nearRT-RIC and xApp can we install now to achieve this control? Do you have any suggestions or comments?

**Reply by P4sca1** [reply]
I only looked into monitoring xApps yet.

**Comment by pgawlowicz** [comment, marked as answer]
Hi @Summer-GY  and @P4sca1 ,

We have just released a repository that allows running ORAN-SC-RIC as a multi-container Docker application with a single docker command. You can access the repository at: https://github.com/srsran/oran-sc-ric

Moreover, the repository includes two Python-based xApps examples designed to facilitate the requesting and monitoring of E2 metrics from the gNB. These examples cover all reporting styles (1-5) of the e2sm-kpm module, which has been updated to version 3.

To get started quickly, refer to the `Quick Start` section in the readme file.
Please note that until the next code release, you'll need to utilize the `test` branch from the srsRAN_Project repository.

We are currently working on an example xApp using e2sm_rc module.

**Reply by yc541** [reply]
Many thanks for the updates, I will have a look at the new repository at some point.

**Reply by P4sca1** [reply]
Thank you for the effort. The ability to deploy the ORAN RIC with a single Docker compose file instead of a Kubernetes cluster, makes experimenting with it a lot easier.
I was able to successfully integrate the repository into my test network. The Python xApp is running successfully.

**Reply by LoganCome** [reply]
Hi @pgawlowicz 

The ORAN-SC-RIC repository and xApps examples are great! They've made experimenting much easier.

Regarding the e2sm_rc module xApp example you mentioned - do you have a rough timeline for when it might be available? We're quite interested.

Thanks for all the excellent work!
